src/operator/tensor/elemwise_binary_op_extended.cc

```diff
@@ -22,76 +22,79 @@
  * \brief CPU Implementation of unary function.
  */
 #include "./elemwise_unary_op.h"
-#include "./elemwise_binary_op.h"
+#include "./elemwise_binary_op-inl.h"
 
 namespace mxnet {
 namespace op {
-MXNET_OPERATOR_REGISTER_BINARY(_power)
+MXNET_OPERATOR_REGISTER_BINARY_WITH_SPARSE_CPU_DR(_power, mshadow_op::power)
 .add_alias("_Power")
-.set_attr<FCompute>("FCompute<cpu>", BinaryCompute<cpu, mshadow_op::power>)
 .set_attr<nnvm::FGradient>("FGradient", ElemwiseGradUseIn{"_backward_power"});
 
 NNVM_REGISTER_OP(_backward_power)
 .set_num_inputs(3)
 .set_num_outputs(2)
 .set_attr<nnvm::TIsBackward>("TIsBackward", true)
 .set_attr<nnvm::FInplaceOption>("FInplaceOption",
-  [](const NodeAttrs& attrs){
-    return std::vector<std::pair<int, int> >{{0, 1}};
-  })
-.set_attr<FCompute>("FCompute<cpu>", BinaryBackwardUseIn<cpu, mshadow_op::power_grad,
-                                                              mshadow_op::power_rgrad>);
+                                [](const NodeAttrs &attrs) {
+                                  return std::vector<std::pair<int, int> >{{0, 1}};
+                                })
+.set_attr<FInferStorageType>("FInferStorageType", ElemwiseStorageType<3, 2>)
+.set_attr<FCompute>("FCompute<cpu>", ElemwiseBinaryOp::BackwardUseIn<
+  cpu, mshadow_op::power_grad, mshadow_op::power_rgrad>);
 
-MXNET_OPERATOR_REGISTER_BINARY(_maximum)
+MXNET_OPERATOR_REGISTER_BINARY_WITH_SPARSE_CPU(_maximum, mshadow_op::maximum)
+MXNET_ADD_SPARSE_OP_ALIAS(maximum)
```

Comment:
I prefer to document it clearly that `sparse.maximum` does not perform broadcasting instead of renaming it, considering broadcasting and element-wise maximum would eventually be consolidated for sparse maximum.

Reply:
Another comment on the macro `MXNET_ADD_SPARSE_OP_ALIAS`. I prefer to write `.add_alias(_sparse_op_name)` explicitly than use this macro. The macro does not reflect visually the fact that `MXNET_ADD_SPARSE_OP_ALIAS(op_name)` must follow a line that returns the op object reference.

Reply:
I'll just remove the alias